docs/source/user_guide/interoperability.rst

```diff
@@ -198,6 +198,82 @@ PyTorch also supports zero-copy data exchange through ``DLPack`` (see :ref:`dlpa
    ...     ...
 
 
+Using custom kernels in PyTorch
+-------------------------------
+
+With the DLPack protocol, it becomes very simple to implement functions in pytorch using CuPy user-defined kernels. Below is the example of a PyTorch autograd function
+that computes the forward and backward pass of the logarithm using `cupy.RawKernel` s.
+
+.. code:: python
+
+    import cupy
+    import torch
+    
+    
+    cupy_custom_kernel_fwd = cupy.RawKernel(
+        r"""
+    extern "C" __global__
+    void cupy_custom_kernel_fwd(const float* x, float* y, int size) {
+        int tid = blockDim.x * blockIdx.x + threadIdx.x;
+        if (tid < size)
+            y[tid] = log(x[tid]);
+    }
+    """,
+        "cupy_custom_kernel_fwd",
+    )
+    
+    
+    cupy_custom_kernel_bwd = cupy.RawKernel(
+        r"""
+    extern "C" __global__
+    void cupy_custom_kernel_bwd(const float* x, float* gy, float* gx, int size) {
+        int tid = blockDim.x * blockIdx.x + threadIdx.x;
+        if (tid < size)
+            gx[tid] = gy[tid] / x[tid];
+    }
+    """,
+        "cupy_custom_kernel_bwd",
+    )
+    
+    
+    class CuPyLog(torch.autograd.Function):
+        @staticmethod
+        def forward(ctx, x):
+            ctx.input = x
+            # Enforce contiguous arrays to simplify RawKernel indexing.
+            cupy_x = cupy.ascontiguousarray(cupy.from_dlpack(x.detach()))
```

Comment:
Silly question: Would there be any impact in PyTorch's workflow with `x` being detached here?

Reply:
you can't export tensors with `requires_grad` set, so we need to explicitly detach them, otherwise autograd will not work :D

Reply:
Yeah my question is probably more silly that this, let's chat offline!